server/routerlicious/packages/services-utils/src/redisClientConnectionManager.ts

```diff
@@ -107,18 +121,38 @@ export class RedisClientConnectionManager implements IRedisClientConnectionManag
 			this.redisOptions.clusterRetryStrategy = this.redisOptions.retryStrategy;
 		}
 
+		const redisClusteringOptions: Partial<Redis.ClusterOptions> = {
+			redisOptions: this.redisOptions,
+			slotsRefreshTimeout: this.slotsRefreshTimeout,
+			dnsLookup: (adr, callback) => callback(null, adr),
+			showFriendlyErrorStack: true,
+			retryDelayOnFailover: this.retryDelays.retryDelayOnFailover,
+			retryDelayOnClusterDown: this.retryDelays.retryDelayOnClusterDown,
+			retryDelayOnTryAgain: this.retryDelays.retryDelayOnTryAgain,
+			retryDelayOnMoved: this.retryDelays.retryDelayOnMoved,
+			maxRedirections: this.retryDelays.maxRedirections,
+		};
+
+		// Remove password from the options objects that will be logged
+		const loggableRedisOptions = { ...this.redisOptions };
+		loggableRedisOptions.password = undefined;
+
+		const loggableClusteringOptions = { ...redisClusteringOptions };
+		loggableClusteringOptions.redisOptions = loggableRedisOptions;
+
+		const stringifiedOptions = this.enableClustering
+			? JSON.stringify(loggableClusteringOptions)
+			: JSON.stringify(loggableRedisOptions);
+
+		Lumberjack.verbose(
+			`Creating a redis client with the following configs: ${stringifiedOptions}`,
+		);
+
 		this.client = this.enableClustering
-			? new Redis.Cluster([{ port: this.redisOptions.port, host: this.redisOptions.host }], {
-					redisOptions: this.redisOptions,
-					slotsRefreshTimeout: this.slotsRefreshTimeout,
-					dnsLookup: (adr, callback) => callback(null, adr),
-					showFriendlyErrorStack: true,
-					retryDelayOnFailover: this.retryDelays.retryDelayOnFailover,
-					retryDelayOnClusterDown: this.retryDelays.retryDelayOnClusterDown,
-					retryDelayOnTryAgain: this.retryDelays.retryDelayOnTryAgain,
-					retryDelayOnMoved: this.retryDelays.retryDelayOnMoved,
-					maxRedirections: this.retryDelays.maxRedirections,
-			  })
+			? new Redis.Cluster(
+					[{ port: this.redisOptions.port, host: this.redisOptions.host }],
+					redisClusteringOptions,
+			  )
 			: new Redis.default(this.redisOptions);
 		Lumberjack.info("Redis client created");
```

Comment:
Nit: Would it make sense to add stringifiedOptions to this log line? Also, would it be useful to log this.enableClustering too?

Reply:
I think that makes sense. We had originally talked about this options log being a verbose log, but I think it is ok to add to this info log. I think it makes sense in this case to add it as a lumber property

Reply:
Updated